test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/CaseSensitiveTagHelperAttributeDescriptorComparer.cs

```diff
@@ -24,13 +24,17 @@ public override bool Equals(TagHelperAttributeDescriptor descriptorX, TagHelperA
                 return true;
             }
 
+            // Normal comparer doesn't care about case, in tests we do. Also double-check IsStringProperty though
+            // it is inferred from TypeName.
             return base.Equals(descriptorX, descriptorY) &&
-                // Normal comparer doesn't care about case, in tests we do.
+                descriptorX.IsStringProperty == descriptorY.IsStringProperty &&
```

Comment:
If the ctor change for `TagHelperAttributeDescriptor` is done, you wont need to do this.